gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/pooled_grpc_asyncio.py.j2

```diff
@@ -0,0 +1,445 @@
+        # Return the channel from cache.
+        channel = self.channel_pool[self._next_idx]
+        self._next_idx = (self._next_idx + 1) % len(self.channel_pool)
+        return channel
+
+    async def replace_channel(self, channel_idx, grace=None, new_channel=None) -> aio.Channel:
+        """
+        Replaces a channel in the pool with a fresh one.
+
+        The `new_channel` will start processing new requests immidiately,
+        but the old channel will continue serving existing clients for `grace` seconds
+
+        Args:
+          channel_idx(int): the channel index in the pool to replace
+          grace(Optional[float]): The time to wait until all active RPCs are 
+            finished. If a grace period is not specified (by passing None for
+            grace), all existing RPCs are cancelled immediately.
+          new_channel(grpc.aio.Channel): a new channel to insert into the pool
+            at `channel_idx`. If `None`, a new channel will be created.
+        """
+        if channel_idx >= len(self.channel_pool) or channel_idx < 0:
+            raise ValueError(
+                f"invalid channel_idx {channel_idx} for pool size {len(self.channel_pool)}"
+            )
+        if new_channel is None:
+            new_channel = self.create_channel(
+                self._host,
+                credentials=self._credentials,
+                credentials_file=None,
+                scopes=self._scopes,
+                ssl_credentials=self._ssl_channel_credentials,
+                quota_project_id=self._quota_project_id,
+                options=[
+                    ("grpc.max_send_message_length", -1),
+                    ("grpc.max_receive_message_length", -1),
+                ],
+            )
+        old_channel = self.channel_pool[channel_idx]
+        self.channel_pool[channel_idx] = new_channel
+        await old_channel.close(grace=grace)
+        # invalidate stubs
+        stub_keys = list(self._stubs.keys())
+        for stub_channel, stub_func in stub_keys:
+            if stub_channel == old_channel:
+                del self._stubs[(stub_channel, stub_func)]
+        return new_channel
+
+    {% if service.has_lro %}
+
+    @property
+    def operations_client(self) -> operations_v1.OperationsAsyncClient:
+        """Create the client designed to process long-running operations.
+
+        This property caches on the instance; repeated calls return the same
+        client.
+        """
+        # Quick check: Only create a new client if we do not already have one.
+        if self._operations_client is None:
+            self._operations_client = operations_v1.OperationsAsyncClient(
+                self.grpc_channel
+            )
+
+        # Return the client from cache.
+        return self._operations_client
+    {% endif %}
+    {% for method in service.methods.values() %}
+
+    def {{ method.transport_safe_name|snake_case }}(self, *args, **kwargs) -> Awaitable[{{ method.output.ident }}]:
```

Comment:
In the regular transport these are properties returning callables: https://github.com/googleapis/gapic-generator-python/blob/3bd2f8703e4a1a67e6a4c281890c4f99eda13fe7/gapic/templates/%25namespace/%25name_%25version/%25sub/services/%25service/transports/grpc_asyncio.py.j2#L260

But the return type here is Awaitable (while the docstring still says this returns Callable).

Reply:
I changed the implementation, so this is out of date, but if you're interested in the explaination, the reason for this is that I was running into issues with caching. 

The client would grab a grpc callable here once at init time, and use it to make future calls. But we wanted it to rotate calls between channels, so I needed to make sure it was actually calling my wrapper, and not just the unary callable directly. The signatures look different, but they should actually be the same, since I dropped the `@propery`. (A method with signature X should be functionally equivalent to a property that returns a function with signature X)

But TL;DR, it got a little ugly for my liking, and I moved the pooling implementation to a channel wrapper, which should make maintenance a bit easier